pennylane_qiskit/vqe_runtime/vqe_runner.py

```diff
@@ -0,0 +1,495 @@
+        "description": "Dictionnary containing: "
+        "The number of evaluation at current optimization step."
+        "Parameter vector at current optimization step."
+        "Function value at the current optimization step."
+        "The size of the step.",
+        "type": "dict",
+    }
+
+    provider = IBMQ.get_provider(hub=hub, group=group, project=project)
+
+    program_id = provider.runtime.upload_program(
+        data="pennylane_runtime/vqe_runtime.py", metadata=meta
+    )
+    return program_id
+
+
+def delete_vqe_runner(provider, program_id):
+    """Delete the desired program on IBMQ platform.
+    Args:
+        provider (object): IBMQ provider.
+        program_id (str): Id of the qiskit runtime to be deleted.
+    """
+    provider.runtime.delete_program(program_id)
+
+
+def vqe_runner(
+    backend,
+    hamiltonian,
+    x0,
+    program_id,
+    ansatz="EfficientSU2",
+    ansatz_config=None,
+    optimizer="SPSA",
+    optimizer_config=None,
+    shots=8192,
+    use_measurement_mitigation=False,
+    **kwargs,
+):
+    """Routine that executes a given VQE problem via the sample-vqe program on the target backend.
+
+    Args:
+        backend (ProgramBackend): Qiskit backend instance.
+        hamiltonian (qml.Hamiltonian): Hamiltonian whose ground state we want to find.
+        x0 (array_like): Initial vector of parameters.
+        program_id(str): Id of the program, it has to be generated by using the upload_vqe_runner function.
+        Once the program is uploaded, you can find the id in your program list online.
+        ansatz (Quantum function or str): Optional, a PennyLane quantum function or the name of the Qiskit
+            ansatz quantum circuit to use, default='EfficientSU2'
+        ansatz_config (dict): Optional, configuration parameters for the ansatz circuit if from Qiskit library.
+        optimizer (str): Optional, string specifying classical optimizer, default='SPSA'.
+        optimizer_config (dict): Optional, configuration parameters for the optimizer.
+        shots (int): Optional, number of shots to take per circuit, default=1024.
+        use_measurement_mitigation (bool): Optional, use measurement mitigation, default=False.
+
+    Returns:
+        OptimizeResult: The result in SciPy optimization format.
+    """
+    # Init the dictionnaries
+    if ansatz_config is None:
+        ansatz_config = {}
+
+    if optimizer_config is None:
+        optimizer_config = {"maxiter": 100}
+
+    if not isinstance(hamiltonian, qml.Hamiltonian):
+        raise qml.QuantumFunctionError("A PennyLane Hamiltonian object is required.")
+
+    connect(kwargs)
+
+    options = {"backend_name": backend}
+
+    inputs = {}
+
+    # Extract the number of qubit from the hamiltonian
+    _, observables = hamiltonian.terms
+
+    qubit_set = set()
+    for obs in observables:
+        for qubit in obs.wires.tolist():
+            qubit_set.add(qubit)
+
+    num_qubits_h = list(qubit_set)[-1] + 1
```

Comment:
The previous thread seems to have gone away.

> This is the intended behavior, because the circuit might not be empty where the hamiltonian does not act. The number of qubit in the circuit is 10 in your example. But I might choose a better variable name..

How do we then determine where we have gates applied? Do we simply just create a 10 qubit circuit regardless of where gates are acting?

Reply:
I changed the logic behind the number oof qubits :)